cmd/bosun/conf/rule/rule.go

```diff
@@ -436,333 +436,19 @@ var defaultFuncs = ttemplate.FuncMap{
-			a.Warn = c.NewExpr(v)
-		case "depends":
-			a.Depends = c.NewExpr(v)
-		case "squelch":
-			a.RawSquelch = append(a.RawSquelch, v)
-			if err := a.Squelch.Add(v); err != nil {
-				c.error(err)
-			}
-		case "critNotification":
-			procNotification(v, a.CritNotification)
-		case "warnNotification":
-			procNotification(v, a.WarnNotification)
-		case "unknown":
-			od, err := opentsdb.ParseDuration(v)
-			if err != nil {
-				c.error(err)
-			}
-			d := time.Duration(od)
-			if d < time.Second {
-				c.errorf("unknown duration must be at least 1s")
-			}
-			a.Unknown = d
-		case "maxLogFrequency":
-			od, err := opentsdb.ParseDuration(v)
-			if err != nil {
-				c.error(err)
-			}
-			d := time.Duration(od)
-			if d < time.Second {
-				c.errorf("max log frequency must be at least 1s")
-			}
-			a.MaxLogFrequency = d
-		case "unjoinedOk":
-			a.UnjoinedOK = true
-		case "ignoreUnknown":
-			a.IgnoreUnknown = true
-		case "unknownIsNormal":
-			a.UnknownsNormal = true
-		case "log":
-			a.Log = true
-		case "runEvery":
-			var err error
-			a.RunEvery, err = strconv.Atoi(v)
-			if err != nil {
-				c.error(err)
-			}
-		default:
-			c.errorf("unknown key %s", p.key)
-		}
-	}
-	if a.MaxLogFrequency != 0 && !a.Log {
-		c.errorf("maxLogFrequency can only be used on alerts with `log = true`.")
-	}
-	c.at(s)
-	if a.Crit == nil && a.Warn == nil {
-		c.errorf("neither crit or warn specified")
-	}
-	var tags eparse.Tags
-	var ret models.FuncType
-	if a.Crit != nil {
-		ctags, err := a.Crit.Root.Tags()
-		if err != nil {
-			c.error(err)
-		}
-		tags = ctags
-		ret = a.Crit.Root.Return()
-	}
-	if a.Warn != nil {
-		wtags, err := a.Warn.Root.Tags()
-		if err != nil {
-			c.error(err)
-		}
-		wret := a.Warn.Root.Return()
-		if a.Crit == nil {
-			tags = wtags
-			ret = wret
-		} else if ret != wret {
-			c.errorf("crit and warn expressions must return same type (%v != %v)", ret, wret)
-		} else if !tags.Equal(wtags) {
-			c.errorf("crit tags (%v) and warn tags (%v) must be equal", tags, wtags)
-		}
-	}
-	if a.Depends != nil {
-		depTags, err := a.Depends.Root.Tags()
-		if err != nil {
-			c.error(err)
-		}
-		if len(depTags) != 0 && len(depTags.Intersection(tags)) < 1 {
-			c.errorf("Depends and crit/warn must share at least one tag.")
-		}
-	}
-	warnLength := len(a.WarnNotification.Notifications) + len(a.WarnNotification.Lookups)
-	critLength := len(a.CritNotification.Notifications) + len(a.CritNotification.Lookups)
-	if a.Log {
-		for _, n := range a.CritNotification.Notifications {
-			if n.Next != nil {
-				c.errorf("cannot use log with a chained notification")
-			}
-		}
-		for _, n := range a.WarnNotification.Notifications {
-			if n.Next != nil {
-				c.errorf("cannot use log with a chained notification")
-			}
-		}
-		if warnLength+critLength == 0 {
-			c.errorf("log specified but no notification")
+	"makeSlice": func(vals ...interface{}) interface{} {
+		return vals
+	},
+	"makeMap": func(vals ...interface{}) (interface{}, error) {
```

Comment:
Need to think, but avoid return errors here because we don't wan't to completely break the template. See the docs at http://bosun.org/definitions#template-error-handling

Somewhat related, we might need an "error" template type for fallback with templates error... same case with JSON.